syntax = "proto3";

import "forch/proto/shared_constants.proto";

message Device {
  string mac = 1;
}

message DevicePortEvent {
  PortBehavior.PortState state = 1;
  int32 device_vlan = 3;
  int32 assigned_vlan = 4;
}

message DevicesState {
  map<string, DevicePlacement> device_mac_placements = 1;
  map<string, DeviceBehavior> device_mac_behaviors = 2;
}

message DevicePlacement {
  string switch = 1;
  int32 port = 2;
  bool connected = 3;
}

message DeviceBehavior {
  string segment = 1;
  string role = 2;
  oneof behavior {
    PortBehavior.Behavior port_behavior = 3;
    PortBehavior.AutoSequestering auto_sequestering = 4;
  }
}

message SegmentsToVlans {
  map<string, int32> segments_to_vlans = 1;
}
